#!/usr/bin/env python3

# # I have neither given nor received any unauthorizedaid in completing this work,
# # nor have I presented someone else's work as my own.
# # Your Name: Devson Butani
# # LTU ID: 000732711
# # Date: 12/13/2022

PACKAGE = "term_project_pkg"  # Catkin Package Name

# Import library
from dynamic_reconfigure.parameter_generator_catkin import *

# Create Instance
gen = ParameterGenerator()

# LIDAR range limits (values outside range are ignored)
# 0 is back of bot, increases CCW, ie: 180 is front center

gen.add(
    "lidar_limit_left",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Range on the LEFT (180 Center, 270 Left)",  # display on hover
    265,  # default
    180,  # min
    360,  # max
)

gen.add(
    "lidar_limit_right",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Range on the RIGHT (180 Center, 90 Right)",  # display on hover
    95,  # default
    0,  # min
    180,  # max
)

gen.add(
    "turning_clearance",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Turning circle clearance of the Robot to turn in front of obstacles",  # display on hover
    350,  # default
    350,  # min
    1000,  # max
)

gen.add(
    "squeeze_through_width",  # variable name
    double_t,  # data type
    0,  # msg level
    ">>> Width of the Robot + Clearance (mm) to drive through a gap",  # display on hover
    470,  # default
    400,  # min
    600,  # max
)

gen.add(
    "pause_detect_obstacle",  # variable name
    bool_t,  # data type
    0,  # msg level
    ">>> Pauses Object Detection for Debug",  # display on hover
    False,  # default
)

# Generate the menu in rqt tool
exit(gen.generate(PACKAGE, "detect_obstacle", "DetectObstacle"))
# 1st param - Package name constant from top
# 2nd param - node name this could be called in (for documentation only)
# 3rd param - This file's name without extension. Adds 'Config' at the end
#           - Use term_project_pkg.cfg and SnowBlowerConfig in py script

# for details: https://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
